lib/src/migrators/division.dart

```diff
@@ -244,40 +272,45 @@ class _DivisionMigrationVisitor extends MigrationVisitor {
       return false;
     }
 
-    ListExpression? channels;
-    if (node.arguments.positional.length == 1 &&
-        node.arguments.named.isEmpty &&
-        node.arguments.positional.first is ListExpression) {
-      channels = node.arguments.positional.first as ListExpression?;
-    } else if (node.arguments.positional.isEmpty &&
-        node.arguments.named.containsKey(r'$channels') &&
-        node.arguments.named.length == 1 &&
-        node.arguments.named[r'$channels'] is ListExpression) {
-      channels = node.arguments.named[r'$channels'] as ListExpression?;
-    }
-    if (channels == null ||
-        channels.hasBrackets ||
-        channels.separator != ListSeparator.space ||
-        channels.contents.length != 3 ||
-        channels.contents.last is! BinaryOperationExpression) {
-      return false;
-    }
+    var channels = switch (node.arguments) {
+      ArgumentInvocation(positional: [ListExpression arg, ...], named: Map()) =>
+        arg,
+      ArgumentInvocation(
+        positional: [],
+        named: {r'$channels': ListExpression arg}
+      ) =>
+        arg,
+      _ => null,
+    };
 
-    var last = channels.contents.last as BinaryOperationExpression;
-    if (last.left is! NumberExpression || last.right is! NumberExpression) {
-      // Handles cases like `rgb(10 20 30/2 / 0.5)`, since converting `30/2` to
-      // `divide(30, 20)` would cause `/ 0.5` to be interpreted as division.
-      _patchSpacesToCommas(channels);
-      _patchOperatorToComma(last);
+    if (channels
+        case ListExpression(
+          hasBrackets: false,
+          separator: ListSeparator.space,
+          contents: [_, _, BinaryOperationExpression last]
+        )) {
+      // Handles cases like `rgb(10 20 30/2 / 0.5)`, since converting `30/2`
+      // to `div(30, 20)` would cause `/ 0.5` to be interpreted as division.
+      switch (last) {
+        case BinaryOperationExpression(
+            left: NumberExpression _,
+            right: NumberExpression _
+          ):
+          break;
+        default:
+          _patchSpacesToCommas(channels);
+          _patchOperatorToComma(last);
+      }
```

Comment:
This could just be:

```suggestion
      if (last case BinaryOperationExpression(
        left: NumberExpression(),
        right: NumberExpression()
      )) {
        break;
      }
      _patchSpacesToCommas(channels);
      _patchOperatorToComma(last);
```

Reply:
The break was to break out of the switch, since we only want to call the `_patch*` functions when the pattern is _not_ matched, but we do still want to do the `_withContext` part either way. Is this structure (with an empty `if` and the patch calls in an `else`) better than the switch?

Reply:
Oh, I see. In Dart 3 you no longer need a `break` at the end of a `case` block; they don't fall through unless either the block is empty or you explicitly opt into it with `continue`.

In this case I'd like `if`/`case`/`else`.

Reply:
What is the idiomatic way to have an empty `case` block that doesn't fall through?

Reply:
In _that_ case you need `break`, but only that case.